modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/dag/schema.scala

```diff
@@ -0,0 +1,95 @@
+package org.tessellation.rosetta.server.model.dag
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import enumeratum.{Enum, EnumEntry}
+import io.circe.Encoder
+import io.estatico.newtype.macros.newtype
+
+/** Represents schema changes specific to this project related to metadata or type Any fields which do not properly code generate
+  */
+object schema {
+
+  case class GenericMetadata()
+
+  @derive(eqv, show)
+  sealed trait ChainObjectStatus extends EnumEntry
+
+  object ChainObjectStatus extends Enum[ChainObjectStatus] with ChainObjectStatusCodecs {
+    case object Unknown extends ChainObjectStatus
+    case object Pending extends ChainObjectStatus
+    case object Accepted extends ChainObjectStatus
+
+    override def values: IndexedSeq[ChainObjectStatus] = findValues
+  }
+
+  trait ChainObjectStatusCodecs {
+    implicit val jsonEncoder: Encoder[ChainObjectStatus] =
+      Encoder.forProduct1("status")(_.toString)
+  }
+
+  @newtype
+  case class NetworkChainObjectStatus(value: ChainObjectStatus)
+
+  case class AccountBalanceResponseMetadata(sequenceNum: Int)
+
+  case class AccountCoinsResponseMetadata(sequenceNum: Int)
+
+  case class AccountIdentifierMetadata(alias: Option[String])
+
+  case class AmountMetadata()
```

Comment:
empty `case class` should be a `case object`

Reply:
There's a few empty case classes here that are part of the dynamically typed Rosetta schema. This isn't actually used anymore, as I only tried to start capturing the ones that we might need in the future (everything else is left under GenericMetadata for simplicity during code-gen.)

I don't remember where this is actually intended to be used, would have to check the Rosetta docs, I'm assuming as one of the metadata fields on the Amount object which might vary between different chains. We don't actually need this code immediately, but it's one of the 'special' metadata fields we'll need to customize. So it has no purpose as a case object anyways, in the future we'd need to populate this with additional custom fields in order to do customization related to different currencies (I think, something like that.) I left this here to distinguish it because there's a hundred different objects and this is one of the few we need to fill out in the future.